services/director/src/simcore_service_director/registry_proxy.py

```diff
@@ -375,69 +375,91 @@ def get_service_last_names(image_key: str) -> str:
     return service_last_name
 
 
+COMPOSE_SPEC_ENTRY_NAME = "ComposeSpec".lower()
+RESOURCES_ENTRY_NAME = "Resources".lower()
+
+
+def _validate_kind(entry_to_validate: Dict[str, Any], kind_name: str):
+    for element in (
+        entry_to_validate.get("value", {})
+        .get("Reservations", {})
+        .get("GenericResources", [])
+    ):
+        if element.get("DiscreteResourceSpec", {}).get("Kind") == kind_name:
+            return True
+    return False
+
+
 async def get_service_extras(
     app: web.Application, image_key: str, image_tag: str
 ) -> Dict[str, Any]:
-    result = {}
-    labels = await get_image_labels(app, image_key, image_tag)
-    logger.debug("Compiling service extras from labels %s", pformat(labels))
-
     # check physical node requirements
     # all nodes require "CPU"
-    result["node_requirements"] = {
-        "CPU": config.DEFAULT_MAX_NANO_CPUS / 1.0e09,
-        "RAM": config.DEFAULT_MAX_MEMORY,
+    result = {
+        "node_requirements": {
+            "CPU": config.DEFAULT_MAX_NANO_CPUS / 1.0e09,
+            "RAM": config.DEFAULT_MAX_MEMORY,
+        }
     }
-    # check if the service requires GPU support
 
-    def validate_kind(entry_to_validate, kind_name):
-        for element in (
-            entry_to_validate.get("value", {})
-            .get("Reservations", {})
-            .get("GenericResources", [])
-        ):
-            if element.get("DiscreteResourceSpec", {}).get("Kind") == kind_name:
-                return True
-        return False
+    labels = await get_image_labels(app, image_key, image_tag)
+    logger.debug("Compiling service extras from labels %s", pformat(labels))
 
+    # check if the service requires GPU support
```

Comment:
are you sure about that comment?

Reply:
That comment was already there and I guess I missed it.

In any case, this module needs a complete refactoring. I just patched it up to make it work. A clean sdk to access the registry shall replace this